lib/aur-chroot--standalone

```diff
@@ -0,0 +1,346 @@
+#!/bin/bash
+# aur-chroot - build packages with systemd-nspawn
+[[ -v AUR_DEBUG ]] && set -o xtrace
+set -o errexit
+argv0=chroot
+PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
+PS4='+(${BASH_SOURCE}:${LINENO}): ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
+machine=$(uname -m)
+
+# default arguments
+directory=/var/lib/aurbuild/$machine
+makepkg_args=()
+makepkg_user=build
+
+# default options
+update=0 build=0 create=0 status=0 run_namcap=0
+
+args_csv() {
+    # shellcheck disable=SC2155
+    local str=$(printf '%s,' "$@")
+    printf '%s' "${str%,}"
+}
+
+# XXX: a missing makepkg.conf usually indicates a missing devtools, whereas
+# a missing pacman.conf usually indicates the local repository was not configured
+diag_makepkg_conf() {
+    echo >&2 'Error:'
+
+    cat <<EOF | pr -to 4 >&2
+aur-$argv0 could not find a makepkg.conf(5) file for container usage. Before
+using aur-$argv0, make sure this file is created and valid. See OPTIONS in
+aur-$argv0(1) for configuration details.
+
+The following file paths were checked:
+EOF
+    printf '%s\n' "${@@Q}" | pr -to 8 >&2
+}
+
+diag_pacman_conf() {
+    echo >&2 'Error:'
+
+    cat <<EOF | pr -to 4 >&2
+aur-$argv0 could not find a pacman.conf(5) file for container usage. Before
+using aur-$argv0, make sure this file is created and valid. See OPTIONS in
+aur-$argv0(1) for configuration details.
+
+The following file paths were checked:
+EOF
+    printf '%s\n' "${@@Q}" | pr -to 8 >&2
+}
+
+usage() {
+    printf 'usage: %s [-BU] [--create] [-CDM path] [package...]\n' "$argv0"
+    exit 1
+}
+
+opt_short='C:D:M:x:ABNTU'
+opt_long=('directory:' 'pacman-conf:' 'makepkg-conf:' 'build' 'update'
+          'create' 'bind:' 'bind-rw:' 'user:' 'makepkg-args:'
+          'ignorearch' 'namcap' 'checkpkg' 'temp' 'makechrootpkg-args:'
+          'margs:' 'cargs:' 'nocheck' 'suffix:')
+opt_hidden=('dump-options' 'status')
+
+if opts=$(getopt -o "$opt_short" -l "$(args_csv "${opt_long[@]}" "${opt_hidden[@]}")" -n "$argv0" -- "$@"); then
+    eval set -- "$opts"
+else
+    usage
+fi
+
+# _chroot functions are "jitted" in a script inside the chroot and executed there
+# in order to run complex commands and keep things tidy
+# idea comes from https://gitlab.archlinux.org/archlinux/devtools/-/blob/master/src/makechrootpkg.in
+
+_chroot_create() {
+    # XXX: use localedef to generate locales directly (outside of nspawn)
+    locale-gen
+
+    pacman-key --init
+    pacman-key --populate
+}
+
+# https://gitlab.archlinux.org/archlinux/devtools/-/blob/master/src/makechrootpkg.in?ref_type=heads#L235
+_chroot_run_namcap() {
+    sudo pacman -S --needed --noconfirm namcap
+    for pkgfile in PKGBUILD /pkgdest/*; do
+        namcap "$pkgfile" 2>&1 | tee "/logdest/${pkgfile##*/}-namcap.log"
```

Comment:
`makechrootpkg` also uses the build user to run namcap, with latter requiring no additional privileges.

Reply:
I think this is what we do here... do I have to change anything?

Reply:
Wouldn't `namcap` run as root here?